src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOp.java

```diff
@@ -821,12 +814,16 @@ private boolean haveNullCase() {
             return false;
         }
 
+        private boolean isSelectorOfTypePrimitive() {
+            return List.of(CHAR, BYTE, SHORT, INT).contains(operands().get(0).type());
+        }
+
         @Override
         public Block.Builder lower(Block.Builder b, OpTransformer opT) {
 
             Value selectorExpression = b.context().getValue(operands().get(0));
 
-            if (!haveNullCase()) {
+            if (!isSelectorOfTypePrimitive() && !haveNullCase()) {
```

Comment:
I think you can replace `!(selectorExpression.type() instance of PrimitiveType)` ?